### PR TITLE
Add version argument -v,--version

### PR DESCRIPTION
**Pull request by jgehrig**
A basic implementation from `QCommandlLineParser` providing the neovim-qt version.

Version information is essential when debugging user issues, and we should provide the standard `-v` and `--version` command-line interface for printing this information.

Example:
```
> nvim-qt --version
nvim-qt 0.2.14.0
```

### CODE REVIEW

**Comment by justinmk** [comment]
perhaps concatenate this with the result of `nvim -v` , then user reports are likely to have that.

**Comment by jgehrig** [comment]
> perhaps concatenate this with the result of nvim -v , then user reports are likely to have that.

I was thinking this too... I like the idea, and want to add the feature.

The QCommandLineParser parser being used is rather limited. I am not sure if QCommandLineParser::showVersion() supports the customization needed for something like this.

File: src/gui/app.cpp [hunk 1]
```diff
@@ -201,6 +207,10 @@ void App::processCliOptions(QCommandLineParser &parser, const QStringList& argum
 		parser.showHelp();
 	}
 
+	if (parser.isSet("version")) {
+		parser.showVersion();
```
Comment (equalsraf) [inline src/gui/app.cpp]: Is this check and the call to showVersion necessary? Reading the docs it seemed the call to QCommandLineParser::process would handle this case.

**Comment by equalsraf** [comment]
> I was thinking this too... I like the idea, and want to add the feature.

It looks like the "easy" way is to add it to the existing version

```c++
setApplicationVersion(PROJECT_VERSION + nvim_version_output);
```

I wonder if checking for

>  	if (parser.isSet("version")) {
>          setApplicationVersion(PROJECT_VERSION + nvim_version_output);

is correct to avoid calling nvim when --version is not used.

Failing that i think we can remove ` 	parser.addVersionOption();` and replace it with our custom code.

```diff
diff --git a/src/gui/app.cpp b/src/gui/app.cpp
index ccf1d92..fa5207e 100644
--- a/src/gui/app.cpp
+++ b/src/gui/app.cpp
@@ -7,6 +7,7 @@
 #include <QTextStream>
 #include "mainwindow.h"
 #include "arguments.h"
+#include "version.h"
 
 namespace NeovimQt {
 
@@ -184,6 +185,8 @@ void App::processCliOptions(QCommandLineParser &parser, const QStringList& argum
 				QCoreApplication::translate("main", "addr")));
 	parser.addOption(QCommandLineOption("spawn",
 				QCoreApplication::translate("main", "Treat positional arguments as the nvim argv")));
+	parser.addOption(QCommandLineOption("version",
+				QCoreApplication::translate("main", "Print the version and exit")));
 	parser.addHelpOption();
 
 #ifdef Q_OS_UNIX
@@ -197,6 +200,26 @@ void App::processCliOptions(QCommandLineParser &parser, const QStringList& argum
 
 	parser.process(arguments);
 
+	if (parser.isSet("version")) {
+		QTextStream out(stdout);
+		out << PROJECT_NAME << ": " << PROJECT_VERSION << endl;
+
+		QProcess nvimproc;
+		QStringList args;
+		args << "--version";
+		QString nvim = "nvim";
+		if (parser.isSet("nvim")) {
+			nvim = parser.value("nvim");
+		}
+		nvimproc.start(nvim, args);
+		if (nvimproc.waitForFinished()) {
+			QString nvim_output(nvimproc.readAllStandardOutput());
+			out << nvim << ": "<< nvim_output.replace("\n", "\n  ") << endl;
+		}
+
+		::exit(0);
+	}
+
 	if (parser.isSet("help")) {
 		parser.showHelp();
 	}
```

Which prints

```
$ bin/nvim-qt --version     
neovim-qt: 0.2.14.0
nvim: NVIM v0.4.0-dev
  Build type: RelWithDebInfo
  Lua 5.1
  Compilation: /usr/bin/cc -g -O2 -fdebug-prefix-map=/build/neovim-8J3t9V/neovim-0.4.0+ubuntu4+git201907232023-5cccfa7-6e51973=. -fstack-protector-strong -Wformat -Werror=format-security -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -g -DMIN_LOG_LEVEL=3 -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -I/build/neovim-8J3t9V/neovim-0.4.0+ubuntu4+git201907232023-5cccfa7-6e51973/build/config -I/build/neovim-8J3t9V/neovim-0.4.0+ubuntu4+git201907232023-5cccfa7-6e51973/src -I/build/neovim-8J3t9V/neovim-0.4.0+ubuntu4+git201907232023-5cccfa7-6e51973/.deps/usr/include -I/usr/include -I/build/neovim-8J3t9V/neovim-0.4.0+ubuntu4+git201907232023-5cccfa7-6e51973/build/src/nvim/auto -I/build/neovim-8J3t9V/neovim-0.4.0+ubuntu4+git201907232023-5cccfa7-6e51973/build/include
  Compiled by buildd@lcy01-amd64-010
  
  Features: +acl +iconv +tui 
  See ":help feature-compile"
  
     system vimrc file: "$VIM/sysinit.vim"
    fall-back for $VIM: "/usr/share/nvim"
  
  Run :checkhealth for more info
```

It is kind of strange to respect --nvim here but it does work. Also it is perfectly fine to have nvim-qt in path but have no nvim in PATH, maybe we should print not found instead.

**Comment by jgehrig** [comment]
How about this?

This provides information in similar formats for both nvim and nvim-qt.
```
> nvim-qt -v
NVIM-QT v0.2.14.0
Build type: Debug
Compilation: -Wall -Wextra -Wno-unused-parameter -std=c++11 -Wfatal-errors
Environment: 
  nvim: nvim

NVIM v0.3.7
Build type: Release
LuaJIT 2.0.5
Compilation: /usr/bin/x86_64-pc-linux-gnu-gcc -march=native -O2 -pipe -Wconversion -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DMIN_LOG_LEVEL=3 -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=auto -Wno-array-bounds -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -I/tmp/portage/app-editors/neovim-0.3.7/work/neovim-0.3.7_build/config -I/tmp/portage/app-editors/neovim-0.3.7/work/neovim-0.3.7/src -I/usr/include -I/tmp/portage/app-editors/neovim-0.3.7/work/neovim-0.3.7_build/src/nvim/auto -I/tmp/portage/app-editors/neovim-0.3.7/work/neovim-0.3.7_build/include
Compiled by portage@jgehrig-x1y

Features: +acl +iconv +jemalloc +tui 
See ":help feature-compile"

   system vimrc file: "/etc/vim/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```
```
> build/bin/nvim-qt -v --nvim nvim-bad-exe
NVIM-QT v0.2.14.0
Build type: Debug
Compilation: -Wall -Wextra -Wno-unused-parameter -std=c++11 -Wfatal-errors
Environment: 
  nvim: nvim-bad-exe

Neovim Not Found!
```
It would also be nice to provide a `runtime` field for `NVIM_QT_RUNTIME_PATH`. However, this will require some refactoring.

**Comment by equalsraf** [comment]
Looks good, nice point on the runtime point but i dont mind merging without.

**Comment by jgehrig** [comment]
That sounds good to me.

I will look into adding the runtime path, and post another Pull Request.

**Comment by equalsraf** [comment]
Merged, thanks @jgehrig

**Pull request merged**